Math/NumberTheory/Moduli/CubicSymbol.hs

```diff
@@ -0,0 +1,100 @@
+module Math.NumberTheory.Moduli.CubicSymbol
+    ( CubicSymbol(..)
+    , conj
+    , cubicSymbol
+    ) where
+
+import Math.NumberTheory.Quadratic.EisensteinIntegers
+import qualified Math.NumberTheory.Quadratic.EisensteinIntegers as E
+
+import qualified Math.NumberTheory.Primes as P
+import qualified Data.Euclidean as A
+import Data.Semigroup as S
+
+
+
+data CubicSymbol = Zero | Omega | OmegaSquare | One deriving (Eq)
+
+instance S.Semigroup CubicSymbol where
+    (<>) = multiplication
+
+instance Show CubicSymbol where
+    show x
+        | x == Zero         = show 0
+        | x == Omega        = "ω"
+        | x == OmegaSquare  = "ω²"
+        | x == One          = show 1
```

Comment:
Please enable `{-# LANGUAGE LambdaCase #-}` and use pattern matching, e. g.,
```haskell
show = \case 
  Zero  -> ...
  Omega -> ...
  ...
```

Reply:
Done.